src/snowflake/cli/api/project/project_paths.py

```diff
@@ -9,8 +9,14 @@ class ProjectPaths:
 
     @property
     def deploy_root(self) -> Path:
-        return self.project_root / "output"
+        return deploy_root(self.project_root)
 
     def remove_up_deploy_root(self) -> None:
         if self.deploy_root.exists():
             rmtree(self.deploy_root)
+
+
+def deploy_root(root: Path, app_type: str | None = None) -> Path:
+    if app_type:
+        return root / "output" / "deploy" / app_type
+    return root / "output" / "deploy"
```

Comment:
1. Native apps have made this a configurable property in snowflake.yml since day 1. I think all domains should support that as well.
2. This isn't resilient to having multiple entities of the same type, which is a big deal if we start deploying multiple entities in a dependency chain. I think we should either use `output/deploy`, as native apps do, for all domains now (it's consistent and simple for now), or make `output/deploy/<entity id>` work across the board. That last one is probably significantly bigger in terms of implementation however.

Reply:
1. I will add.
2. You mean to mix artifacts from all entities in `output/deploy`? I added `output/deploy/snowpark` and `output/deploy/streamlit` to separate artifacts for each entity type, to keep similar solution what we have right now.

Reply:
there won't be any mixing, each bundle/deploy step should wipe out `output/deploy`. Segregating by type doesn't buy us anything since multiple entities for the same type could have conflicting artifacts.

Reply:
About #1, I factored out the model portion on our feature branch already.